app/models/category.rb

```diff
@@ -1118,6 +1118,46 @@ def has_restricted_tags?
     tags.count > 0 || tag_groups.count > 0
   end
 
+  def self.reorder_categories!
+    categories =
+      if SiteSetting.fixed_category_positions
+        Category.order(:position)
+      else
+        Category
+          .left_outer_joins(:featured_topics)
+          .group("categories.id")
+          .order("max(topics.bumped_at) DESC NULLS LAST")
+      end.order(:id).pluck(:id, :parent_category_id)
```

Comment:
One problem here is that we are loading everything into memory and can cause memory to bloat.